File: src/Dfe.PlanTech.Domain/Questionnaire/Models/RecommendationIntro.cs
using System.ComponentModel.DataAnnotations.Schema;
using Dfe.PlanTech.Domain.Content.Models;
using Dfe.PlanTech.Domain.Questionnaire.Interfaces;

namespace Dfe.PlanTech.Domain.Questionnaire.Models;

public class RecommendationIntro : ContentComponent, IRecommendationIntro<Header, ContentComponent>, IHeaderWithContent
{
    public string Slug { get; init; } = null!;

    public Header Header { get; init; } = null!;

    public string Maturity { get; init; } = null!;

    [NotMapped]
    public List<ContentComponent> Content { get; init; } = [];

    public string HeaderText => Header.Text;

    public string LinkText => "Overview";

    private string? _slugifiedLinkText;

    public string SlugifiedLinkText => _slugifiedLinkText ??= LinkText.Slugify();
}

File: contentful/export-processor/content-types/recommendation-intro.js
export default class RecommendationIntro {
  content;
  header;
  id;
  maturity;
  slug;

  constructor({ fields, sys }) {
    this.content = fields.content;
    this.header = fields.header;
    this.id = sys.id;
    this.maturity = fields.maturity;
    this.slug = fields.slug;
  }
}
